### ISSUE TITLE
Argument 1 passed to Symfony\Component\Uid\Uuid::isValid() must be of the type string, object given

### ISSUE THREAD

**Issue by tjamps** [Guikingone/SymfonyUidDoctrine#1]
When using this repository to generate an UUID on Doctrine entities, I encounter the following error:

```
In Uuid.php line 101:
                                                                                                                                                                                                        
  [TypeError]                                                                                                                                                                                           
  Argument 1 passed to Symfony\Component\Uid\Uuid::isValid() must be of the type string, object given, called in /var/www/html/vendor/guikingone/symfony-uid-doctrine/src/Uuid/UuidType.php on line 45  
                                                                                                                                                                                                        

Exception trace:
  at /var/www/html/vendor/symfony/uid/Uuid.php:101
 Symfony\Component\Uid\Uuid::isValid() at /var/www/html/vendor/guikingone/symfony-uid-doctrine/src/Uuid/UuidType.php:45
 Guikingone\SymfonyUid\Doctrine\Uuid\UuidType->convertToPHPValue() at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:1526
 Doctrine\DBAL\Connection->convertToPHPValue() at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:3625
 Doctrine\ORM\UnitOfWork->convertSingleFieldIdentifierToPHPValue() at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:952
 Doctrine\ORM\UnitOfWork->persistNew() at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:1780
 Doctrine\ORM\UnitOfWork->doPersist() at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php:1736
 Doctrine\ORM\UnitOfWork->persist() at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:609
 Doctrine\ORM\EntityManager->persist() at /var/www/html/src/DataFixtures/SportFixtures.php:19
 App\DataFixtures\SportFixtures->load() at /var/www/html/vendor/doctrine/data-fixtures/lib/Doctrine/Common/DataFixtures/Executor/AbstractExecutor.php:115
 Doctrine\Common\DataFixtures\Executor\AbstractExecutor->load() at /var/www/html/vendor/doctrine/data-fixtures/lib/Doctrine/Common/DataFixtures/Executor/ORMExecutor.php:69
 Doctrine\Common\DataFixtures\Executor\ORMExecutor::Doctrine\Common\DataFixtures\Executor\{closure}() at n/a:n/a
 call_user_func() at /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:239
 Doctrine\ORM\EntityManager->transactional() at /var/www/html/vendor/doctrine/data-fixtures/lib/Doctrine/Common/DataFixtures/Executor/ORMExecutor.php:71
 Doctrine\Common\DataFixtures\Executor\ORMExecutor->execute() at /var/www/html/vendor/doctrine/doctrine-fixtures-bundle/Command/LoadDataFixturesDoctrineCommand.php:127
 Doctrine\Bundle\FixturesBundle\Command\LoadDataFixturesDoctrineCommand->execute() at /var/www/html/vendor/symfony/console/Command/Command.php:258
 Symfony\Component\Console\Command\Command->run() at /var/www/html/vendor/symfony/console/Application.php:929
 Symfony\Component\Console\Application->doRunCommand() at /var/www/html/vendor/symfony/framework-bundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /var/www/html/vendor/symfony/console/Application.php:264
 Symfony\Component\Console\Application->doRun() at /var/www/html/vendor/symfony/framework-bundle/Console/Application.php:82
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /var/www/html/vendor/symfony/console/Application.php:140
 Symfony\Component\Console\Application->run() at /var/www/html/bin/console:45
```

The `$value` parameter passed to `Guikingone\SymfonyUid\Doctrine\Uuid\UuidType::convertToPHPValue` is a `Symfony\Component\Uid\UuidV4`. 

I am using Symfony 5.1.2, and the following versions of Doctrine and your package:

```
doctrine/annotations                 1.10.3  Docblock Annotations Parser
doctrine/cache                       1.10.1  PHP Doctrine Cache library is ...
doctrine/collections                 1.6.5   PHP Doctrine Collections libra...
doctrine/common                      2.13.3  PHP Doctrine Common project is...
doctrine/data-fixtures               1.4.3   Data Fixtures for all Doctrine...
doctrine/dbal                        2.10.2  Powerful PHP database abstract...
doctrine/doctrine-bundle             2.1.0   Symfony DoctrineBundle
doctrine/doctrine-fixtures-bundle    3.3.1   Symfony DoctrineFixturesBundle
doctrine/doctrine-migrations-bundle  3.0.1   Symfony DoctrineMigrationsBundle
doctrine/event-manager               1.1.0   The Doctrine Event Manager is ...
doctrine/inflector                   1.4.3   PHP Doctrine Inflector is a sm...
doctrine/instantiator                1.3.1   A small, lightweight utility t...
doctrine/lexer                       1.2.1   PHP Doctrine Lexer parser libr...
doctrine/migrations                  3.0.1   PHP Doctrine Migrations projec...
doctrine/orm                         v2.7.3  Object-Relational-Mapper for PHP
doctrine/persistence                 1.3.7   The Doctrine Persistence proje...
doctrine/reflection                  1.2.1   The Doctrine Reflection projec...
doctrine/sql-formatter               1.1.0   a PHP SQL highlighting library
guikingone/symfony-uid-doctrine      0.1.0   Allow the use of symfony/uid a...
symfony/doctrine-bridge              v5.1.2  Symfony Doctrine Bridge
```
I've never written a custom Doctrine type, but I'm enclined to help you on this one. What do you think is the best approach to solve that issue ?

**Comment by Guikingone** [comment]
Hi @tjamps 

That's actually ... a bug 😄 

I've opened a PR here: https://github.com/Guikingone/SymfonyUidDoctrine/pull/2

Feel free to review and tell me if it solve the issue, this way, I can create a new release 🙂

**Comment by andrejlola** [comment]
Hi.
The fix solve the issue.

**Comment by Guikingone** [comment]
Bug fix merged and version 0.1.1 released, thanks for the issue @andrejlola 🙂

**Comment by tjamps** [comment]
I can confirm the issue is fixed with your patch (sorry for the delay).

Thanks a lot !
